packages/terra-data-grid/src/FlowsheetDataGrid.jsx

```diff
@@ -247,18 +258,18 @@ function FlowsheetDataGrid(props) {
   }, [rows, columns, onCellRangeSelect]);
 
   const handleCellSelection = useCallback((selectionDetails) => {
-    // Exclude the row header column.
+    // Call onRowSelect for row header column
     if (selectionDetails.columnIndex === 0) {
-      return;
-    }
-
-    if (selectionDetails.isShiftPressed && anchorCell.current !== null) {
+      if (onRowSelect) {
+        onRowSelect(selectionDetails.rowId);
```

Comment:
Now that we have section support (understanding it is a work in progress), shouldn't we return both the section id and row id?

Reply:
I can add the section id as a param here. Would this also apply to onCellSelect and onCellRangeSelect?

Reply:
I think we have a separate JIRA logged for the onCellSelect and onCellRangeSelect work.  That will be a little tricky as we will have to determine if we want to break the existing API and make the section id the first parameter.  We probably should have been returning objects instead of a parameter list for this reason.

Reply:
Maybe onRowSelect should "return" an object as well?  I am open to opinions.

Reply:
I would be fine with making it an object. That way it might be a bit easier on consumers if we were to add anything to it, rather than introducing a new parameter.

Reply:
Changed to an object